assertj-core/src/main/java/org/assertj/core/api/AbstractStringAssert.java

```diff
@@ -402,4 +408,279 @@ public SELF isEqualTo(String expectedStringTemplate, Object... args) {
+   * <p>
+   * Note that only when the string is equal to the string "true", ignoring case, and can have leading and trailing space, the parsed value will be {@code true}.
+   * Other not null strings will make the value be {@code false}.
+   * <p>
+   * Examples:
+   * <pre><code class='java'> // assertion succeeds
+   * assertThat(&quot;truE&quot;).asBoolean().isTrue();
+   * assertThat(&quot;false&quot;).asBoolean().isFalse();
+   * assertThat(&quot;foo bar&quot;).asBoolean().isFalse();
+   *
+   * // assertion fails as the actual value is null
+   * assertThat((String) null).asBoolean();</code></pre>
+   *
+   * @return a new {@link BooleanAssert} instance whose value under test is the result of the parse.
+   * @throws AssertionError if the actual value is {@code null}.
+   *
+   * @since 3.24.0
+   */
+  public AbstractBooleanAssert<?> asBoolean() {
+    return super.extracting(s -> Boolean.parseBoolean(s.trim()), InstanceOfAssertFactories.BOOLEAN);
+  }
+
+  /**
+   * Parses the actual value as byte, using radix 10, the parsed byte becoming the new value under test.
+   * <p>
+   * Examples:
+   * <pre><code class='java'> // assertion succeeds
+   * assertThat(&quot;127&quot;).asByte().isEqualTo((byte) 127);
+   *
+   * // assertion fails as the actual value is null or not a byte
+   * assertThat((String) null).asByte();</code></pre>
+   * assertThat(&quot;1L&quot;).asByte();</code></pre>
+   *
+   * @return a new {@link ByteAssert} instance whose value under test is the result of the parse.
+   * @throws AssertionError if the actual value is {@code null}.
+   * @throws AssertionError if the actual value is not numeric.
+   * @throws AssertionError if the actual value is out of the range of byte.
+   *
+   * @since 3.24.0
+   */
+  public AbstractByteAssert<?> asByte() {
+    return asByte(10);
+  }
+
+  /**
+   * Parses the actual value as byte, using the specified radix, the parsed byte becoming the new value under test.
+   * <p>
+   * Examples:
+   * <pre><code class='java'> // assertion succeeds
+   * assertThat(&quot;100&quot;).asByte(2).isEqualTo((byte) 4);
+   *
+   * // assertion fails as the actual value is null or not parseable
+   * assertThat((String) null).asByte(2);</code></pre>
+   * assertThat(&quot;9&quot;).asByte(2);</code></pre>
+   *
+   * @param radix the radix to be used while parsing actual value as a byte.
+   * @return a new {@link ByteAssert} instance whose value under test is the result of the parse.
+   * @throws AssertionError if the actual value is {@code null}.
+   * @throws AssertionError if the actual value is not numeric.
+   * @throws AssertionError if {@code radix} is either smaller than {@code Character.MIN_RADIX} or larger than {@code Character.MAX_RADIX}.
+   * @throws AssertionError if the actual value (under the specified radix) is out of the range of byte.
+   *
+   * @since 3.24.0
+   */
+  public AbstractByteAssert<?> asByte(int radix) {
```

Comment:
I'm not convinced of the need of exposing any conversion with radix, let's keep the API small until somebody asks for it.
I have rarely used parsing with a radix as a java developer so I don't expect it to be a prevalent use for assertions.